Tab details conversion in request details page from HAML to React

mahantidinesh7 opened the pull request:
Before :
<img width="1728" alt="Screenshot 2023-04-03 at 1 30 21 PM" src="https://user-images.githubusercontent.com/61232160/229450390-f537931b-5b3d-491c-81a2-a1e40fbb9723.png">
After :
<img width="1727" alt="Screenshot 2023-04-03 at 1 32 09 PM" src="https://user-images.githubusercontent.com/61232160/229450417-c7398113-1808-4e22-888b-54e82f657d50.png">

Note : This PR is specific only for tab details conversions and the other conversion in this page have been already handled in
this PR https://github.com/ManageIQ/manageiq-ui-classic/pull/8731 .

@miq-bot assign @MelsHyrule

jeffibm commented on app/helpers/st_prov_details_helper.rb:
```diff
@@ -0,0 +1,77 @@
+  end
+
+  def st_prov_show_tab_configuration(work_flow)
+    tab_labels = []
+    work_flow.dialog.dialog_tabs.each_with_index do |tab, _tab_index|
+      tab_labels.push(tab_label(tab))
+    end
+    tab_labels
```
```
 work_flow.dialog.dialog_tabs.map do |tab|
     tab_labels.push(tab_label(tab))
  end
```

jeffibm commented on app/helpers/st_prov_details_helper.rb:
```diff
@@ -0,0 +1,77 @@
+    value = ''
+    case field.type
+    when 'DialogFieldTextBox', 'DialogFieldTextAreaBox'
+      value = if field.protected?
```
`value = ifield.protected? ? '********' : field.value`

mahantidinesh7 commented on app/helpers/st_prov_details_helper.rb:
```diff
@@ -0,0 +1,73 @@
+  end
+
+  def st_prov_show_tab_configuration(work_flow)
+    tab_labels = []
+    work_flow.dialog.dialog_tabs.map do |tab|
+      tab_labels.push(tab_label(tab))
+    end
+    tab_labels
```
I didnt get why we only call this method `tab_label(tab)` and dont push it to tab_labels

jeffibm replied:
`.map` will return an array

jeffibm commented:
Hey @mahantidinesh7 , could you please have a look into the failing specs

mahantidinesh7 commented:
Resolved the issue.  used only one `?` in this `value = classifications.empty? ? '' : classifications.join(', ')` which caused the TC to failure

jeffibm commented:
@mahantidinesh7 , could you rebase this?

mahantidinesh7 commented:
@jeffibm rebased this PR. @MelsHyrule can we merge this PR if everything looks good.its opened since last week

jeffibm commented:
Hey @mahantidinesh7, please have a look at the failing specs

mahantidinesh7 commented:
@jeffibm i have pulled the latest master changes which should fix the failing  specs . please review this PR

mahantidinesh7 commented:
Mybad . It fails for some other test cases. Let me check on the Test cases for them

mahantidinesh7 commented:
@jeffibm  we have to remove the old test cases as they have been written for HAML which we have removed while moving to react. I have written new test cases in react which cover the same test cases as that of HAML and also rebased the branch. please review this PR

jeffibm commented on app/views/miq_request/_st_prov_show.html.haml:
```diff
@@ -11,22 +11,11 @@
+            - catalog_tab_labels = st_prov_show_tab_configuration(wf)            
+            = react('MiqCustomTab', {:containerId => 'st-prov-show-tabs-content',
+                          :tabLabels => catalog_tab_labels,
+                          :type => 'CATALOG_SUMMARY'})
```
I think you need to add a new type here. maybe `PROV_DETAILS` will do. This is to identify where these tabs are used.

jeffibm commented on app/helpers/st_prov_details_helper.rb:
```diff
@@ -0,0 +1,69 @@
+        end
+      end
+      summary.push(miq_structured_list(
+                     :title => _(group.label),
```
@DavidResende0 , could you please check if the translations work here?

DavidResende0 replied:
By default, just applying _() to a variable will not actually mark the string for translation since the check for which strings need translations occurs before any code in the frontend is run.

The solution is to ensure that the actual string in the backend is surrounded by `N_()` which saves the translated string to the db but doesn't actually translate it until _() is called in the frontend later.

From what I can tell though, this string seems like it is generated by the user when they create their dialog. If that's the case then it shouldn't be translated anyways. Not 100% sure though if it is user generated.

jeffibm commented:
Hey @mahantidinesh7,  this has been open for too long. if there is nothing else, could you please rebase so that we can merge this.

jeffibm commented:
Hey @mahantidinesh7 , when you have time, please rebase this and lets merge this..
